hazelcast/src/test/java/com/hazelcast/jet/core/JobTest.java

```diff
@@ -787,6 +791,63 @@ public void when_suspendedNamedJob_then_newJobWithEqualNameFails() {
                 .isInstanceOf(JobAlreadyExistsException.class);
     }
 
+    @Test
+    public void given_suspensionIsForbidden_when_suspendJob_then_jobIsTerminated() {
+        // Given
+        DAG dag = new DAG().vertex(new Vertex("test", new MockPS(NoOutputSourceP::new, NODE_COUNT * 2)));
+        JobConfig jobConfig = new JobConfig();
+        jobConfig.setArgument(JobConfigArguments.KEY_JOB_IS_SUSPENDABLE, false);
+
+        // When
+        Job job = instance().getJet().newJob(dag, jobConfig);
+
+        // Then
+        assertJobStatusEventually(job, RUNNING);
+        assertThatThrownBy(job::suspend)
+                .hasMessageContaining("Cannot suspend the job being analyzed");
+
+        cancelAndJoin(job);
+    }
+
+    @Test
+    public void given_suspensionIsForbidden_when_restartJob_then_jobIsTerminated() {
+        // Given
+        DAG dag = new DAG().vertex(new Vertex("test", new MockPS(NoOutputSourceP::new, NODE_COUNT * 2)));
+        JobConfig jobConfig = new JobConfig();
+        jobConfig.setArgument(JobConfigArguments.KEY_JOB_IS_SUSPENDABLE, false);
+
+        // When
+        Job job = instance().getJet().newJob(dag, jobConfig);
+
+        // Then
+        assertJobStatusEventually(job, RUNNING);
+        assertThatThrownBy(job::restart)
```

Comment:
error message is misleading for `restart`

Reply:
Javadoc says 
```
 Conceptually this call is equivalent to {@link #suspend()} & {@link
     * #resume()}. Not supported for light jobs.
```